### DISCUSSION TITLE
Migration from laminas-console to laminas-cli

### DISCUSSION THREAD

**Discussion by jjwdesign** [laminas/laminas-cli#99] [Q&A]
Is there any migration guide? I'm trying to migrate from laminas-console (old ZF2 framework) to laminas-cli (latest Laminas). I've noticed that this is significantly different than just setting up a Controller and setting a few config settings.

**Comment by gsteel** [comment]
`laminas-cli` integrates Symfony Console based commands into Laminas and Mezzio applications. It takes care of setting up the console application so that you only have to [create your app-specific commands](https://symfony.com/doc/current/console.html#creating-a-command) and register them under application configuration. I don't believe there is a migration guide because the two packages have very little in common.

Also see: #85

**Reply by froschdesign** [reply]
I will add a migration path and a better quick start guide.

> I've noticed that this is significantly different than just setting up a Controller and setting a few config settings.

And now you create a command and add some configuration settings. Some other steps but not much more either.

**Reply by jjwdesign** [reply]
> I don't believe there is a migration guide because the two packages have very little in common.

Well, it's being recommended by packagist.org.


laminas/laminas-console
Build console applications using getopt syntax or routing, complete with prompts

Abandoned! See [laminas/laminas-cli](https://packagist.org/packages/laminas/laminas-cli)

I just need a simple way of executing a Controller via the command line. We used to do this with laminas-console, which was very easy to setup. We have an existing ConsoleController with a few methods in our Module. Am I correct in attempting to use laminas/laminas-cli as a replacement?

Thank you.

**Reply by gsteel** [reply]
> I just need a simple way of executing a Controller via the command line

`laminas-cli` does not provide a simple way of executing a controller. If it is important to retain this functionality, then you can continue to use `laminas-console`.

If you have kept your controllers thin and practised dependency injection, the general approach would be to setup a new command _(based on Symfony console)_ that accepts your services and models to its constructor and performs the desired operations, create a factory to instantiate it with all of its required dependencies and finally, you'd add it to your application configuration as outlined in the Readme and execute it from the cli using `vendor/bin/laminas your:command`

**Reply by jjwdesign** [reply]
> > I just need a simple way of executing a Controller via the command line
> 
> `laminas-cli` does not provide a simple way of executing a controller. If it is important to retain this functionality, then you can continue to use `laminas-console`.

Unfortunately, I can not continue to use laminas-console for two reasons. When migrating to PHP8+ it's not supported. Also, our security team/person doesn't like abandoned packages.

But, I did finally get it to work. I had some path issues and some little details to take care of. I'll start another discussion to see what you guys think about the issues I hit and if there's any better solution(s) going forward.

Thank you for the package!
